charmhelpers/contrib/database/mysql.py

```diff
@@ -78,9 +78,11 @@ def __init__(self, rpasswdf_template, upasswdf_template, host='localhost',
         self.delete_ondisk_passwd_file = delete_ondisk_passwd_file
         self.connection = None
 
-    def connect(self, user='root', password=None):
-        log("Opening db connection for %s@%s" % (user, self.host), level=DEBUG)
-        self.connection = MySQLdb.connect(user=user, host=self.host,
+    def connect(self, user='root', password=None, host=None):
+        if not host:
```

Comment:
As `host` is None in this test, the correct python code is:

```python
if host is not None:
    host = self.host
```

Reply:
This now needs a test, as it's overriding the `self.host`, and thus the contract needs to be enforced in the future.  Thanks.

Reply:
Hi Alex,

Shouldn't it be:

```
if host is None:
    host = self.host
```

so that it can either be specified or defaults to  `self.host` if not specified?

Reply:
@tytus-kurek yes, sorry, that's a typo on my part.  Well caught!  Whilst these are equivalent:

```python
if not host:
    ...

if host is None:
    ...
```

The latter is more correct if we are using `None` as a sentinel to indicate that a parameter has not been passed.